src/calendar/tests/CalendarMonth.spec.js

```diff
@@ -0,0 +1,232 @@
+    });
+    this.firstOfMonth = moment();
+  });
+
+  it('should render the right element', function () {
+    this.useShallowRenderer();
+    expect(this.renderedComponent.type).toBe('div');
+    expect(this.spyCx).toHaveBeenCalledWith({
+      element: 'Month',
+    });
+    expect(this.renderedComponent.props.className).toEqual('Month');
+  });
+
+  describe('has a label acting as a header', function () {
+
+    beforeEach(function () {
+      this.useShallowRenderer();
+    });
+
+    it('which is a div with the correct class', function () {
+      expect(this.container.type).toBe('div');
+      expect(this.container.props.className).toEqual('MonthHeader');
+      expect(this.spyCx).toHaveBeenCalledWith({
+        element: 'MonthHeader',
+      });
+    });
+
+    describe('displaying month information', function () {
+
+      it('which creates a span with the correct class', function () {
+
+        this.useShallowRenderer();
+        const span = this.container.props.children[0];
+
+        expect(span.type).toBe('span');
+        expect(span.props.className).toEqual('MonthHeaderLabel');
+        expect(this.spyCx).toHaveBeenCalledWith({
+          element: 'MonthHeaderLabel',
+          modifiers: {
+            month: true,
+          },
+        });
+      });
+
+      it('which displays the name of the month', function () {
+        this.useShallowRenderer();
+        const span = this.container.props.children[0];
+        expect(span.props.children[0]).toBe(this.firstOfMonth.format('MMMM'));
+      });
+
+      it('which does not show navigation if props.disableNavigation is true', function () {
+        this.useShallowRenderer({
+          disableNavigation: true,
+        });
+        const span = this.container.props.children[0];
+        expect(span.props.children[1]).toBe(null);
+      });
+
+      it('which shows navigation if props.disableNavigation is false', function () {
+        this.useShallowRenderer();
+        const select = this.container.props.children[0].props.children[1];
+        expect(select.type).toBe('select');
+        expect(this.spyCx).toHaveBeenCalledWith({
+          element: 'MonthHeaderSelect',
+        });
+        expect(select.props.value).toBe(this.firstOfMonth.month());
+        expect(select.props.className).toEqual('MonthHeaderSelect');
+        expect(select.props.children.length).toBe(12);
+      });
+
+      it('which calls props.onMonthChange if props.disableNavigation is false and if the selected value changes', function () {
+        var onMonthChange = jasmine.createSpy();
+        this.useDocumentRenderer({
+          onMonthChange: onMonthChange,
+        });
+        var select = TestUtils.scryRenderedDOMComponentsWithTag(this.renderedComponent, 'select')[0].getDOMNode();
+        var value = (select.value === '1') ? '2' : '1';
+        select.value = value;
+        TestUtils.Simulate.change(select);
+        expect(onMonthChange).toHaveBeenCalledWith(parseInt(value, 10));
```

Comment:
I wonder if this should be static / hard coded, as we should know the expectation of our object under test